gcp/gce_instance.go

```diff
@@ -0,0 +1,85 @@
+package gcp
+
+import (
+	"errors"
+	"strings"
+	"time"
+)
+
+func zoneFromUrl(url string) (string, error) {
+	split := strings.Split(url, "/")
+	if len(split) == 0 {
+		return "", errors.New("got invalid zone url: " + url)
+	}
+	return split[len(split)-1], nil
+}
+
+func regionFromZone(ctx *GcpContext, zone string) (string, error) {
```

Comment:
Add unit tests for this method. It's important that we know it works correctly so we don't nuke the wrong zone/region.